source/extensions/common/wasm/context.cc

```diff
@@ -454,6 +454,13 @@ Context::findValue(absl::string_view name, Protobuf::Arena* arena, bool last) co
   switch (part_token->second) {
   case PropertyToken::METADATA:
     if (info) {
+
+      return CelProtoWrapper::CreateMessage(&info->dynamicMetadata(), arena);
+    }
+    break;
+  case PropertyToken::DYNAMICMETADATA:
+    if (info) {
+
       return CelProtoWrapper::CreateMessage(&info->dynamicMetadata(), arena);
```

Comment:
Why do we need another token returning `&info->dynamicMetadata()`? Is there any reason why we cannot re-use `METADATA`?

Reply:
About `setProperty`:
I had to add `DYNAMICMETADATA` to understand when to call [`stream_info->setDynamicMetadata`](  https://github.com/envoyproxy/envoy/pull/15196/files/a7a88acf00ae348a18ba6646ce7f62146d0439e9#diff-c386f59113c79157fdc6b88ce0c4c14e307509bd7038ce816821985293d55410R1147)

So to  `set` `dynamicmetadata` you can do something like:
```rust
self.set_property(vec!["dynamicmetadata"],...
```

to `set` `metadata`: 
```rust 
self.set_property(vec!["metadata", "filter_metadata", "envoy.filters.network.rbac", "something"], Some("value"));
```

I may be missing something, but I could not find another way to distinguish [`filter_state`](https://github.com/envoyproxy/envoy/blob/a7a88acf00ae348a18ba6646ce7f62146d0439e9/source/extensions/common/wasm/context.cc#L1154) from [`dynamicmetadata`](https://github.com/envoyproxy/envoy/blob/a7a88acf00ae348a18ba6646ce7f62146d0439e9/source/extensions/common/wasm/context.cc#L1139)

About `getProperty` yes you are right I just added `DYNAMICMETADATA` to compile the code, I will handle it better.


If you think that in general a new token is not needed let me know. 
Thank you